src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -2,37 +2,1084 @@
+        {
+            00, 09, 01, 10, 13, 21, 02, 29,
+            11, 14, 16, 18, 22, 25, 03, 30,
+            08, 12, 20, 28, 15, 17, 24, 07,
+            19, 27, 23, 06, 26, 05, 04, 31
+        };
+
+        #region PopCount
+
+        /// <summary>
+        /// Returns the population count (number of bits set) of a mask.
+        /// Similar in behavior to the x86 instruction POPCNT.
+        /// </summary>
+        /// <param name="value">The mask.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static int TrailingZeroCount(int matches)
+        public static uint PopCount(uint value)
         {
-            if (Bmi1.IsSupported)
+            if (Popcnt.IsSupported)
+            {
+                return Popcnt.PopCount(value);
+            }
+
+            const uint c0 = 0x_5555_5555;
+            const uint c1 = 0x_3333_3333;
+            const uint c2 = 0x_0F0F_0F0F;
+            const uint c3 = 0x_0101_0101;
+
+            uint val = value;
+
+            val -= (val >> 1) & c0;
+            val = (val & c1) + ((val >> 2) & c1);
+            val = (val + (val >> 4)) & c2;
+            val *= c3;
+            val >>= 24;
+
+            return val;
+        }
+
+        /// <summary>
+        /// Returns the population count (number of bits set) of a mask.
+        /// Similar in behavior to the x86 instruction POPCNT.
+        /// </summary>
+        /// <param name="value">The mask.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static uint PopCount(int value)
+            => PopCount(unchecked((uint)value));
+
+        /// <summary>
+        /// Returns the population count (number of bits set) of a mask.
+        /// Similar in behavior to the x86 instruction POPCNT.
+        /// </summary>
+        /// <param name="value">The mask.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static uint PopCount(ulong value)
+        {
+            if (Popcnt.IsSupported)
+            {
+                if (Popcnt.X64.IsSupported)
+                {
+                    return (uint)Popcnt.X64.PopCount(value);
+                }
+
+                uint hv = (uint)(value >> 32); // High-32
+                uint bv = (uint)value; // Low-32
+
+                uint h = Popcnt.PopCount(hv);
+                uint b = Popcnt.PopCount(bv);
+
+                return h + b;
+            }
+
+            const ulong c0 = 0x_5555_5555_5555_5555;
+            const ulong c1 = 0x_3333_3333_3333_3333;
+            const ulong c2 = 0x_0F0F_0F0F_0F0F_0F0F;
+            const ulong c3 = 0x_0101_0101_0101_0101;
+
+            ulong val = value;
+
+            val -= (value >> 1) & c0;
+            val = (val & c1) + ((val >> 2) & c1);
+            val = (val + (val >> 4)) & c2;
+            val *= c3;
+            val >>= 56;
+
+            return (uint)val;
+        }
+
+        /* Legacy implementations
```

Comment:
I don't think we need these, that is what source control is for :)

It should be easy enough to see by just looking at the deleted code in this PR.

Reply:
I was trying to make it easy to eye-ball, but it will be cleaner without the noise. I will leave the urls in place.

Reply:
Done